src/app.ts

```diff
@@ -116,6 +117,7 @@ const cachingMiddleware = apicache.options({
 
 export interface IAppFactoryParameters {
   env: NodeEnvironment;
+  appInsightsClient?: appInsights.TelemetryClient;
```

Comment:
Why this param is marked as optional?

Reply:
Because the instantiation of `appInsights` client is subsequent to the presence of `APPINSIGHTS_INSTRUMENTATIONKEY`. When the key is missing the client doesn't exist but the backend starts anyway. [link](https://github.com/pagopa/io-backend/pull/715/files#diff-916156c481141bfcd08c6ae40ce10e00R57-R75)